core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -348,8 +347,23 @@ class ZkUtils(zkClientWrap: ZooKeeperClientWrapper,
     brokerIds.map(_.toInt).map(getBrokerInfo(_)).filter(_.isDefined).map(_.get)
   }
 
-  def getLeaderAndIsrForPartition(topic: String, partition: Int):Option[LeaderAndIsr] = {
-    ReplicationUtils.getLeaderIsrAndEpochForPartition(this, topic, partition).map(_.leaderAndIsr)
+  def getLeaderAndIsrForPartition(topic: String, partition: Int): Option[LeaderAndIsr] = {
```

Comment:
maybe we can add this method to KafkaZkClient. We can use getTopicPartitionState to avoid code duplication.

Reply:
Why if we only use it in ZkUtils. This whole class will be removed so not sure we need to rewrite the methods to use KafkaZkClient.

Reply:
looks like getLeaderAndIsrForPartition method is called from couple of test classes.  when we remove these methods, we need to migrate methods to KafkaZkClient  Right?   
I was thinking callers zkUtils methods can migrate to KafkaZkClient methods.  This can be done later.

Reply:
Yeah, I think we can do test cases last. The methods may end up in a test utils class if it's only used in tests.